client/gcp/gcp.go

```diff
@@ -0,0 +1,212 @@
+		// This condition will filter resource like Network.
+		if rawOffer.Category.ResourceFamily != "ApplicationServices" ||
+			// the Gen1 offering is unavailabe
+			strings.Contains(rawOffer.Category.ResourceGroup, "Gen1") ||
+			!rSpecification.Match([]byte(rawOffer.Description)) {
+			continue
+		}
+
+		offerType := getOfferType(rawOffer.Category.ResourceGroup)
+		hourlyUSD := getUSD(rawOffer.PricingInfo[0].PricingExpression.TieredRateList[0].UnitPrice)
+
+		offer := &client.Offer{
+			ID:        incrID,
+			SKU:       rawOffer.ID,
+			OfferType: offerType,
+
+			// All GCP services are charged on demand literal, but we consider commitment for a length as reserved type
+			ChargeType: client.ChargeTypeOnDemand,
+
+			RegionList:  rawOffer.ServiceRegionList,
+			Description: rawOffer.Description,
+
+			CommitmentUSD: 0,
+			HourlyUSD:     hourlyUSD,
+		}
+		incrID++
+
+		if offerType == client.OfferTypeInstance {
+			databaseEngine, CPU, memory := getCPUMemory(rawOffer.Description)
+			instanceType := getInstanceType(rawOffer.Category.ResourceGroup, CPU, memory)
+			payload := &client.OfferInstancePayload{
+				Type:           instanceType,
+				InstanceFamily: instanceType,
+				CPU:            CPU,
+				Memory:         memory,
+				DatabaseEngine: databaseEngine,
+			}
+			offer.InstancePayload = payload
+		}
+
+		if offer.ChargeType == client.ChargeTypeReserved {
+			panic("Implement me")
+		}
+
+		offerList = append(offerList, offer)
+	}
+	return offerList, nil
+}
+
+// getOfferType will extract the OfferType from a given resource group.
+// Possible resource group are:
+// SQLGen2Instance${INSTANE_CODE}, SQLGen2InstanceRAM, SQLGen2InstanceCPU.
+func getOfferType(resourceGroup string) client.OfferType {
+	offerType := client.OfferType(strings.Replace(resourceGroup, "SQLGen2Instances", "", -1))
+	if offerType != "RAM" && offerType != "CPU" {
+		offerType = client.OfferTypeInstance
+	}
+	return offerType
+}
+
+// Regional Instance means High Available Instance, see https://cloud.google.com/sql/docs/postgres/high-availability
+var rSpecification = regexp.MustCompile(`Cloud SQL for ([\S|\s]+): Zonal - (\d+) vCPU \+ (\d+.\d*)GB RAM`)
+
+// getCPUMemory will use a reg-expression to extract the specification expressed in the given description
+// the description should follow the form of "Cloud SQL for ${ENGINE_TYPE}: Zonal - ${NUM_VCPU} vCPU + ${NUM_MEM}GB RAM".
+func getCPUMemory(description string) (databaseEngine client.EngineType, CPU string, memory string) {
+	match := rSpecification.FindStringSubmatch(description)
+	if match[1] == "MySQL" {
+		databaseEngine = client.EngineTypeMySQL
+	} else if match[1] == "PostgreSQL" {
+		databaseEngine = client.EngineTypePostgreSQL
+	}
+	return databaseEngine, match[2], match[3]
+}
+
+// getUSD will return a single value of the price in USD
+func getUSD(unitPrice unitPrice) float64 {
+	unitInt64, _ := strconv.Atoi(unitPrice.Unit)
+	return float64(unitInt64) + unitPrice.Nano/1e9
```

Comment:
Consider adding comments or even examples of unitPrice.Unit and unitPrice.Nano. By reading this without an example, it's a bit confusing to understand the rational.